### PR TITLE
EfficientSAM Model Export

### PR DESCRIPTION
**Pull request by jakmro**
### Summary
This PR introduces the [EfficientSAM](https://github.com/yformer/EfficientSAM) model, along with the instructions for exporting it to Core ML and XNNPACK using ExecuTorch.

Integrating the export-ready EfficientSAM example into ExecuTorch broadens the range of available examples.

### Test plan
- Followed the instructions in `examples/models/efficient_sam/README.md` to export the model to Core ML and XNNPACK.
- Ensured the exported models function as expected.
- Added `test_efficient_sam_export_to_executorch` in `examples/models/test/test_export.py`.

### CODE REVIEW

File: examples/models/efficient_sam/efficient_sam_core/efficient_sam.py [hunk 1]
```diff
@@ -0,0 +1,316 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (iseeyuan) [inline examples/models/efficient_sam/efficient_sam_core/efficient_sam.py:1]: Could you cite the source of this file, as well as the encoder/decoder?

Comment (iseeyuan) [reply examples/models/efficient_sam/efficient_sam_core/efficient_sam.py:1]: Could you also highlight if there are any modifications to make it exportable to ExecuTorch?

**Review by iseeyuan — CHANGES_REQUESTED** [review]
Thank you for adding this example! Please add more inline comments to the source files: cite sources and highlight modifications if there are any.

File: examples/models/efficient_sam/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+This example demonstrates how to export the [EfficientSAM](https://github.com/yformer/EfficientSAM) model to Core ML and XNNPACK using ExecuTorch.
+
+# Instructions
+
+## 1. Setup
+
+Follow the [tutorial](https://pytorch.org/executorch/main/getting-started-setup#) to set up ExecuTorch.
+
+## 2. Exports
```
Comment (iseeyuan) [inline examples/models/efficient_sam/README.md:11]: Are there any performance numbers of those two backend? Would be great to list them here.

File: examples/models/efficient_sam/efficient_sam_core/efficient_sam.py (hunk 1 above)
Comment (jakmro) [reply examples/models/efficient_sam/efficient_sam_core/efficient_sam.py:1]: Yes, of course

File: examples/models/efficient_sam/README.md (hunk 2 above)
Comment (jakmro) [reply examples/models/efficient_sam/README.md:11]: I have just listed some performance numbers.

**Review by iseeyuan — APPROVED** [review]
LGTM! Thanks for updating!

**Comment by jakmro** [comment]
Hi @iseeyuan, what are the next steps? Is there anything else I can do?

**Comment by iseeyuan** [comment]
@jakmro , could you rebase your PR and make sure all CI tests pass? Referring to https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests, if you have write permission, please go ahead and merge it. If you don't have write permission please let me know and I can merge it for you.

**Comment by jakmro** [comment]
@iseeyuan There are still two Facebook Internal CI checks failing, but I don’t have access to the logs. Could you help me understand what’s causing the issue? Thanks!

**Comment by jakmro** [comment]
> @jakmro , could you rebase your PR and make sure all CI tests pass? Referring to https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests, if you have write permission, please go ahead and merge it. If you don't have write permission please let me know and I can merge it for you.

Unfortunately, I also don't have write permission.

**Comment by iseeyuan** [comment]
@jakmro Let me look into it then. I've disabled internal CIs because they are already covered by OSS CI. I'm rebasing it and see if the internal CI failure is gone.

**Comment by iseeyuan** [comment]
@jakmro The PR has been merged. The major issue is from internal tests, where it does not allow accessing external urls. I've disabled the internal tests since they are already covered by OSS CI. Next time when a new model is added, we get all signals from OSS that all contributors can see, and the efficiency should be much better. Thanks for your patience!

**Pull request merged**